### PR TITLE
Merge core refactor into v1.0 feature branch

### PR DESCRIPTION
**Pull request by ankona**
Combine the `core-refactor` feature branch with `mli-feature` in `v1.0` branch

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -127,12 +135,13 @@ jobs:
           LLP=$(cat $SP | grep LD_LIBRARY_PATH | awk '{split($0, array, "="); print array[2]}')
           echo "LD_LIBRARY_PATH=$LLP:$LD_LIBRARY_PATH" >> $GITHUB_ENV
 
-      - name: Run mypy
-        run: |
-          make check-mypy
+      # - name: Run mypy
+      #   run: |
+      #     make check-mypy
```
Comment (MattToast) [inline .github/workflows/run_tests.yml]: Did this get commented out in the merge? I think we still have `mypy` running in `smartsim-core` and just added suppression in the `pyproject.toml`. In theory you should be free to turn it back on!

File: .github/workflows/build_docs.yml [hunk 2]
```diff
@@ -32,6 +32,8 @@ on:
   push:
     branches:
       - develop
+    branches-ignore:
+      - smartsim-refactor
```
Comment (MattToast) [inline .github/workflows/build_docs.yml]: If we are getting rid of this branch, feel free to drop this addition!

File: .github/workflows/changelog.yml [hunk 3]
```diff
@@ -30,6 +30,8 @@ name: enforce_changelog
 
 on:
   pull_request:
+    branches-ignore:
+      - smartsim-refactor
```
Comment (MattToast) [inline .github/workflows/changelog.yml]: This should also be free to drop!

File: .github/workflows/run_tests.yml [hunk 4]
```diff
@@ -163,7 +172,7 @@ jobs:
         run: |
           echo "SMARTSIM_LOG_LEVEL=debug" >> $GITHUB_ENV
           py.test -s --import-mode=importlib -o log_cli=true --cov=$(smart site) --cov-report=xml --cov-config=./tests/test_configs/cov/local_cov.cfg --ignore=tests/full_wlm/ -m ${{ matrix.subset }} ./tests
-
+
```
Comment (MattToast) [inline .github/workflows/run_tests.yml]: Pedantic nit: As long as you are here, can you get rid of the extra white-space that sneaked in here, lol

File: smartsim/_core/_cli/validate.py [hunk 5]
```diff
@@ -78,35 +70,21 @@ def __exit__(
             self._finalizer.detach()  # type: ignore[attr-defined]
 
 
-def execute(
-    args: argparse.Namespace, _unparsed_args: t.Optional[t.List[str]] = None, /
-) -> int:
+def execute(args: argparse.Namespace, _unparsed_args: argparse.Namespace) -> int:
```
Comment (MattToast) [inline smartsim/_core/_cli/validate.py]: Nit: Not sure exactly what happened here with the change in type, but I'm pretty sure `_uparsed_args` should still be a `list[str] | None`. It doesn't _really_ matter since the param is unused, but while your here, can you fix it up?

File: smartsim/_core/control/manifest.py [hunk 6]
```diff
@@ -127,24 +113,16 @@ def _check_names(deployables: t.List[t.Any]) -> None:
     @staticmethod
     def _check_types(deployables: t.List[t.Any]) -> None:
         for deployable in deployables:
-            if not isinstance(deployable, (SmartSimEntity, EntitySequence)):
+            if not isinstance(deployable, SmartSimEntity):
                 raise TypeError(
-                    f"Entity has type {type(deployable)}, not "
-                    + "SmartSimEntity or EntitySequence"
+                    f"Entity has type {type(deployable)}, not " + "SmartSimEntity"
```
Comment (MattToast) [inline smartsim/_core/control/manifest.py:116]: Style nit, while you're here, lol
```suggestion
                    f"Entity has type {type(deployable)}, not SmartSimEntity"
```

File: smartsim/_core/control/job.py [hunk 7]
```diff
@@ -29,8 +29,10 @@
 import typing as t
 from dataclasses import dataclass
 
-from ...entity import EntitySequence, SmartSimEntity
-from ...status import SmartSimStatus
+from smartsim.entity._mock import Mock
```
Comment (MattToast) [inline smartsim/_core/control/job.py]: Nit: `Mock` looks like its unused, can you drop this import while you're here?

File: smartsim/_core/launcher/step/mpi_step.py [hunk 8]
```diff
@@ -133,29 +135,31 @@ def _build_exe(self) -> t.List[str]:
         if self._get_mpmd():
             return self._make_mpmd()
 
-        exe = self.run_settings.exe
-        args = self.run_settings._exe_args  # pylint: disable=protected-access
+        exe = self.entity.exe
+        args = self.entity.exe_args  # pylint: disable=protected-access
```
Comment (MattToast) [inline smartsim/_core/launcher/step/mpi_step.py]: Nit, no longer accessing a protected attr
```suggestion
        args = self.entity.exe_args
```

File: smartsim/_core/launcher/step/mpi_step.py [hunk 9]
```diff
@@ -133,29 +135,31 @@ def _build_exe(self) -> t.List[str]:
...
         return exe + args
 
     def _make_mpmd(self) -> t.List[str]:
         """Build mpiexec (MPMD) executable"""
-        exe = self.run_settings.exe
-        args = self.run_settings._exe_args  # pylint: disable=protected-access
+        exe = self.entity.exe
+        args = self.entity.exe_args  # pylint: disable=protected-access
```
Comment (MattToast) [inline smartsim/_core/launcher/step/mpi_step.py]: ditto here
```suggestion
        args = self.entity.exe_args
```

File: smartsim/_core/launcher/step/mpi_step.py [hunk 10]
```diff
@@ -133,29 +135,31 @@ def _build_exe(self) -> t.List[str]:
...
         cmd = exe + args
 
         for mpmd in self._get_mpmd():
             cmd += [" : "]
             cmd += mpmd.format_run_args()
             cmd += mpmd.format_env_vars()
             cmd += mpmd.exe
-            cmd += mpmd._exe_args  # pylint: disable=protected-access
+            cmd += mpmd.exe_args  # pylint: disable=protected-access
```
Comment (MattToast) [inline smartsim/_core/launcher/step/mpi_step.py]: And here
```suggestion
            cmd += mpmd.exe_args
```

File: smartsim/_core/launcher/step/lsf_step.py [hunk 11]
```diff
@@ -214,8 +216,8 @@ def _build_exe(self) -> t.List[str]:
 
         :return: executable list
         """
-        exe = self.run_settings.exe
-        args = self.run_settings._exe_args  # pylint: disable=protected-access
+        exe = self.entity.exe
+        args = self.entity.exe_args  # pylint: disable=protected-access
```
Comment (MattToast) [inline smartsim/_core/launcher/step/lsf_step.py]: pedantic lint nit:
```suggestion
        args = self.entity.exe_args
```

File: smartsim/_core/launcher/dragon/dragon_launcher.py [hunk 12]
```diff
@@ -327,3 +370,56 @@ def _assert_schema_type(obj: object, typ: t.Type[_SchemaT], /) -> _SchemaT:
...
+    policy = DragonRunPolicy.from_run_args(run_args)
+    return (
+        DragonRunRequestView(
+            exe=exe_,
+            exe_args=args,
+            # FIXME: Currently this is hard coded because the schema requires
+            #        it, but in future, it is almost certainly necessary that
+            #        this will need to be injected by the user or by us to have
+            #        the command execute next to any generated files. A similar
+            #        problem exists for the other settings.
```
Comment (MattToast) [inline smartsim/_core/launcher/dragon/dragon_launcher.py]: Looks like this got fixed, but we left the `FIXME` comment on accident. CAn you remove while you're here?
```suggestion
```

File: smartsim/_core/launcher/step/slurm_step.py [hunk 13]
```diff
@@ -200,18 +207,21 @@ def _build_exe(self) -> t.List[str]:
...
         """Build Slurm multi-prog (MPMD) executable"""
-        exe = self.run_settings.exe
-        args = self._get_exe_args_list(self.run_settings)
+        exe = self.entity.exe
+        args = self._get_exe_args_list(self.entity)
         cmd = exe + args
 
         compound_env_vars = []
-        for mpmd_rs in self._get_mpmd():
+        for mpmd_rs in self._get_mpmd():  # returns a list of runsettings
```
Comment (MattToast) [inline smartsim/_core/launcher/step/slurm_step.py]: Just trying to remove some potential "tech-debt-y" type comments, while we are here!
```suggestion
        for mpmd_rs in self._get_mpmd():
```

File: smartsim/_core/mli/infrastructure/control/dragon_util.py [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+# isort: on
+
+from smartsim.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def function_as_dragon_proc(
+    entrypoint_fn: t.Callable[[t.Any], None],
+    args: t.List[t.Any],
```
Comment (MattToast) [inline smartsim/_core/mli/infrastructure/control/dragon_util.py:52]: Typing Nit: Probably not worth shoving into this PR, but thoughts on updating the type hints here?
```suggestion
_Ts = TypeVarTuple("_Ts")

def function_as_dragon_proc(
    entrypoint_fn: t.Callable[[Unpack[_Ts]], None],
    args: tuple[Unpack[_Ts]],
```

Comment (MattToast) [reply smartsim/_core/mli/infrastructure/control/dragon_util.py:52]: If we agree this is "out of scope" totally wiling to shove this comment into a new (short task) ticket

File: smartsim/_core/utils/helpers.py [hunk 15]
```diff
@@ -27,47 +27,99 @@
...
+import sys
 import typing as t
 import uuid
+import warnings
 from datetime import datetime
-from functools import lru_cache
 from pathlib import Path
 from shutil import which
 
+from deprecated import deprecated
```
Comment (MattToast) [inline smartsim/_core/utils/helpers.py]: Minor nit: Did `Deprecated` get added to our deps in the `setup.py` or equivalent? I couldn't find it with a naive grep and am slightly worried its coming in through an optional dep (Torch, TF, Dragon, etc.)

Comment (MattToast) [reply smartsim/_core/utils/helpers.py]: Looks like it's actually required by `PyGithub`, so we should be okay, but if you wouldn't mind adding it to the list anyway, just to avoid the transient dep that would be much appreciated!

File: doc/changelog.md [hunk 16]
```diff
@@ -13,6 +13,7 @@ Jump to:
 
 Description
 
+- Merge core refactor into MLI feature branch
```
Comment (amandarichardsonn) [inline doc/changelog.md:16]: The core team did not keep an updated changelog but I notice that the MLI team has, will this be an issue when merged into main and shows on the CrayLabs doc page? I think we could later just add a single line to the develop portion of the changelog to document core updates, this is just a difference I am raising

File: pyproject.toml [hunk 17]
```diff
@@ -134,6 +141,31 @@ module = [
   "smartsim.ml.torch.*",            # must solve/ignore inheritance issues
   "watchdog",
   "dragon.*",
+
+  # Ignore these modules while the core refactor is on going. Uncomment as
+  # needed for gradual typing
+  #
+  # FIXME: DO NOT MERGE THIS INTO DEVELOP BRANCH UNLESS THESE ARE PASSING OR
```
Comment (amandarichardsonn) [inline pyproject.toml:148]: Me not knowing Q: when do we plan to run mypy against these modules?

File: doc/changelog.md (hunk 16 above)
Comment (amandarichardsonn) [inline doc/changelog.md:16]: Im not able to comment on this file but doc/installation_instructions/basic.rst line 48 I believe "availble" is misspelled

Comment (amandarichardsonn) [reply doc/changelog.md:16]: Grammatical error in the same file on line 49 -> “the versions backwards compatible” should be “the versions are backwards compatible”

Comment (amandarichardsonn) [reply doc/changelog.md:16]: And I think "Additional" should be "Additionally"

Comment (amandarichardsonn) [reply doc/changelog.md:16]: In the same file: "succesfully" -> "successfully"

Comment (amandarichardsonn) [reply doc/changelog.md:16]: At the end of the same file -> `"Step 2: Build SmartSim above"` - move above outside of the quotes

File: smartsim/_core/launcher/step/alps_step.py [hunk 18]
```diff
@@ -122,15 +125,15 @@ def _build_exe(self) -> t.List[str]:
         if self._get_mpmd():
             return self._make_mpmd()
 
-        exe = self.run_settings.exe
-        args = self.run_settings._exe_args  # pylint: disable=protected-access
+        exe = self.entity.exe
+        args = self.entity.exe_args  # pylint: disable=protected-access
```
Comment (MattToast) [inline smartsim/_core/launcher/step/alps_step.py]: lint nit:
```suggestion
        exe = self.entity.exe
        args = self.entity.exe_args
```

File: smartsim/_core/launcher/step/alps_step.py [hunk 19]
```diff
@@ -122,15 +125,15 @@ def _build_exe(self) -> t.List[str]:
...
         return exe + args
 
     def _make_mpmd(self) -> t.List[str]:
         """Build Aprun (MPMD) executable"""
 
-        exe = self.run_settings.exe
-        exe_args = self.run_settings._exe_args  # pylint: disable=protected-access
+        exe = self.entity.exe
+        exe_args = self.entity._exe_args  # pylint: disable=protected-access
```
Comment (MattToast) [inline smartsim/_core/launcher/step/alps_step.py]: And here
```suggestion
        exe = self.entity.exe
        exe_args = self.entity.exe_args
```

File: smartsim/settings/common.py [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+def set_check_input(key: str, value: t.Optional[str]) -> None:
+    if not isinstance(key, str):
+        raise TypeError(f"Key '{key}' should be of type str")
+    if not isinstance(value, (str, type(None))):
+        raise TypeError(f"Value '{value}' should be of type str or None")
+    if key.startswith("-"):
+        key = key.lstrip("-")
+        logger.warning(
+            "One or more leading `-` characters were provided to the run argument. \
+            Leading dashes were stripped and the arguments were passed to the run_command."
```
Comment (MattToast) [inline smartsim/settings/common.py]: I'm guessing this message was supposed to read
```suggestion
            "One or more leading `-` characters were provided to the run argument.\n"
            "Leading dashes were stripped and the arguments were passed to the run_command."
```
to avoid the multiple spaces at the start of the next line. Can you update while we are here?

File: tests/_legacy/test_helpers.py [hunk 21]
```diff
@@ -30,12 +30,32 @@
...
+    job_3 = Job(app, launch_settings)
+    job_4 = Job(app, launch_settings)
+    job_5 = Job(app, launch_settings)
+
+    assert (
+        [job_1, job_2, job_3, job_4, job_5]
+        == list(unpack([job_1, [job_2, job_3], job_4, [job_5]]))
+        == list(unpack([job_1, job_2, [job_3, job_4], job_5]))
+        == list(unpack([job_1, [job_2, [job_3, job_4], job_5]]))
+    )
```
Comment (MattToast) [inline tests/_legacy/test_helpers.py:56]: @MattToast this test looks like it is new for the V1 API (and most of the tests in this file look like they are still valid).

TODO: Make a ticket to move valid tests from this file out of `tests/_legacy`

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (amandarichardsonn) [inline .github/workflows/run_tests.yml]: does mypy need to be turned on? I remember Julia is fixing the lint errors

File: Makefile [hunk 22]
```diff
@@ -196,4 +196,4 @@ test-wlm:
 # help: test-dragon                   - Run dragon-specific tests
 .PHONY: test-dragon
 test-dragon:
```
Comment (amandarichardsonn) [inline Makefile:198]: Why do we ignore the dragon test suite for all make targets besides the custom dragon make target?

File: smartsim/experiment.py [hunk 23]
```diff
@@ -182,653 +165,235 @@ def _set_dragon_server_path(self) -> None:
...
-        output_format: previewrenderer.Format = previewrenderer.Format.PLAINTEXT,
+        verbosity_level: preview_renderer.Verbosity = preview_renderer.Verbosity.INFO,
+        output_format: preview_renderer.Format = preview_renderer.Format.PLAINTEXT,
         output_filename: t.Optional[str] = None,
     ) -> None:
         """Preview entity information prior to launch. This method
         aggregates multiple pieces of information to give users insight
         into what and how entities will be launched.  Any instance of
-        ``Model``, ``Ensemble``, or ``Orchestrator`` created by the
+        ``Model``, ``Ensemble``, or ``Feature Store`` created by the
```
Comment (amandarichardsonn) [inline smartsim/experiment.py:396]: ``Model`` -> ``Application``

File: tests/_legacy/test_run_settings.py [hunk 24]
```diff
@@ -567,3 +569,55 @@ def test_update_env_null_valued(env_vars):
...
+    exp = Experiment(exp_name, launcher=wlmutils.get_test_launcher(), exp_path=test_dir)
+    settings = exp.create_run_settings(
+        "echo", exe_args=["hello", "world"], run_command="srun"
+    )
+    settings.set("--nodes", 1)
+
+    model = exp.create_model("sr_issue_model", run_settings=settings)
+    exp.start(model)
+    statuses = exp.get_status(model)
+    assert all([stat == JobStatus.COMPLETED for stat in statuses])
```
Comment (MattToast) [inline tests/_legacy/test_run_settings.py:623]: Sanity checking: Do we know where these tests came from? It looks like they are using the V0.8 API, so I don't think they came from `smartsim-refactor`. Just wanted to make sure they are not important to the MLI work since they ended up in the `tests/_legacy` dir which is not run in the CI (but I'm guessing they are not, they seem orthogonal at best).

Comment (MattToast) [reply tests/_legacy/test_run_settings.py:623]: Actually, on second thought, these probably came from `develop` as it moved beyond `smartsim-refactor`and `mli-feature`. Should be safe where they are now!

**Review by MattToast — CHANGES_REQUESTED** [review]
Couple of super small linting nits, but otherwise it looks like the majority of the `smartsim-core` work is here and in-tack! Thanks for handling this super large merge!!

File: doc/changelog.md (hunk 16 above)
Comment (ankona) [reply doc/changelog.md:16]: I agree that we should change it when we merge v1.0->develop. this is just to make the wheels turn without changing the build pipeline itself. i'll add a ticket to make sure the final changelog is clean

File: pyproject.toml (hunk 17 above)
Comment (ankona) [reply pyproject.toml:148]: IMO, we should get this as-is into v1.0 and work on making mypy pass to get v1.0 into develop. I will add a ticket to avoid losing track of it

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (ankona) [reply .github/workflows/run_tests.yml]: @juliaputko can you add re-enabling this check to your task for fixing these issues

File: smartsim/_core/_cli/validate.py (hunk 5 above)
Comment (ankona) [reply smartsim/_core/_cli/validate.py]: IIRC, it wasn't there at all and the execute was blowing up due to receiving too many arguments. I added it by hand and put the wrong typehint.

File: smartsim/_core/launcher/step/mpi_step.py (hunk 8 above)
Comment (ankona) [reply smartsim/_core/launcher/step/mpi_step.py]: @juliaputko i think you'll fix this

File: smartsim/_core/control/job.py (hunk 7 above)
Comment (ankona) [reply smartsim/_core/control/job.py]: @juliaputko i think you'll fix this

File: smartsim/_core/launcher/step/mpi_step.py (hunk 9 above)
Comment (ankona) [reply smartsim/_core/launcher/step/mpi_step.py]: @juliaputko i think you'll fix this

File: smartsim/_core/launcher/step/mpi_step.py (hunk 10 above)
Comment (ankona) [reply smartsim/_core/launcher/step/mpi_step.py]: @juliaputko i think you'll fix this

File: smartsim/_core/launcher/step/lsf_step.py (hunk 11 above)
Comment (ankona) [reply smartsim/_core/launcher/step/lsf_step.py]: @juliaputko i think you'll fix this

File: smartsim/_core/launcher/step/alps_step.py (hunk 18 above)
Comment (ankona) [reply smartsim/_core/launcher/step/alps_step.py]: @juliaputko i think you'll fix this

File: smartsim/_core/_cli/validate.py (hunk 5 above)
Comment (ankona) [reply smartsim/_core/_cli/validate.py]: fixed!

File: .github/workflows/build_docs.yml (hunk 2 above)
Comment (ankona) [reply .github/workflows/build_docs.yml]: fixed!

File: .github/workflows/changelog.yml (hunk 3 above)
Comment (ankona) [reply .github/workflows/changelog.yml]: fixed!

File: smartsim/_core/launcher/step/alps_step.py (hunk 19 above)
Comment (MattToast) [reply smartsim/_core/launcher/step/alps_step.py]: To be resolved in [this ticket](https://github.com/CrayLabs/SmartSim/issues/762)

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (ankona) [reply .github/workflows/run_tests.yml]: re-enabled

File: smartsim/_core/launcher/step/alps_step.py (hunk 18 above)
Comment (MattToast) [reply smartsim/_core/launcher/step/alps_step.py]: To be resolved in [this ticket](https://github.com/CrayLabs/SmartSim/issues/762)

File: smartsim/_core/launcher/step/lsf_step.py (hunk 11 above)
Comment (MattToast) [reply smartsim/_core/launcher/step/lsf_step.py]: To be resolved in [this ticket](https://github.com/CrayLabs/SmartSim/issues/762)

File: smartsim/experiment.py (hunk 23 above)
Comment (ankona) [reply smartsim/experiment.py:396]: fixed!

File: smartsim/_core/launcher/step/mpi_step.py (hunk 10 above)
Comment (MattToast) [reply smartsim/_core/launcher/step/mpi_step.py]: To be resolved in [this ticket](https://github.com/CrayLabs/SmartSim/issues/762)

File: smartsim/_core/launcher/step/mpi_step.py (hunk 9 above)
Comment (MattToast) [reply smartsim/_core/launcher/step/mpi_step.py]: To be resolved in [this ticket](https://github.com/CrayLabs/SmartSim/issues/762)

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (ankona) [reply .github/workflows/run_tests.yml]: added back. julia is on linter, not pylint

File: smartsim/_core/launcher/step/mpi_step.py (hunk 8 above)
Comment (MattToast) [reply smartsim/_core/launcher/step/mpi_step.py]: To be resolved in [this ticket](https://github.com/CrayLabs/SmartSim/issues/762)

File: smartsim/_core/launcher/dragon/dragon_launcher.py (hunk 12 above)
Comment (ankona) [reply smartsim/_core/launcher/dragon/dragon_launcher.py]: removed

File: smartsim/_core/control/job.py (hunk 7 above)
Comment (MattToast) [reply smartsim/_core/control/job.py]: Will be caught in [this ticket](https://github.com/CrayLabs/Poseidon/issues/830)

File: smartsim/_core/control/manifest.py (hunk 6 above)
Comment (MattToast) [reply smartsim/_core/control/manifest.py:116]: Will be caught in [this ticket](https://github.com/CrayLabs/Poseidon/issues/830)

File: smartsim/_core/utils/helpers.py (hunk 15 above)
Comment (ankona) [reply smartsim/_core/utils/helpers.py]: my plan to remove those before anybody saw them failed. removed.

File: smartsim/_core/launcher/step/alps_step.py (hunk 19 above)
Comment (ankona) [reply smartsim/_core/launcher/step/alps_step.py]: fixed

File: smartsim/settings/common.py (hunk 20 above)
Comment (ankona) [reply smartsim/settings/common.py]: fixed

File: smartsim/_core/launcher/step/alps_step.py (hunk 18 above)
Comment (ankona) [reply smartsim/_core/launcher/step/alps_step.py]: fixed anyway

File: smartsim/_core/launcher/step/mpi_step.py (hunk 8 above)
Comment (ankona) [reply smartsim/_core/launcher/step/mpi_step.py]: fixed already

File: .github/workflows/run_tests.yml (hunk 4 above)
Comment (ankona) [reply .github/workflows/run_tests.yml]: gone

File: smartsim/_core/mli/infrastructure/control/dragon_util.py (hunk 14 above)
Comment (MattToast) [reply smartsim/_core/mli/infrastructure/control/dragon_util.py:52]: Deemed "out-of-scope". Moved to [new ticket](https://github.com/CrayLabs/SmartSim/issues/763)

File: doc/changelog.md (hunk 16 above)
Comment (ankona) [reply doc/changelog.md:16]: fixed `availble`

Comment (ankona) [reply doc/changelog.md:16]: fixed `are bw compat`

Comment (ankona) [reply doc/changelog.md:16]: fixed `additionally`

Comment (ankona) [reply doc/changelog.md:16]: fixed `successfully`

Comment (ankona) [reply doc/changelog.md:16]: moved "above"

File: Makefile (hunk 22 above)
Comment (ankona) [reply Makefile:198]: ~~To ensure that tests requiring the innards of dragon have dragon installed. we might be able to expand where we install dragon later... @ankona create a follow-up ticket to see if we can expand where we install dragon during tests~~

Reviewed and it looks ok. We do execute dragon tests on every platform in the matrix, already. 

- Dragon tests require a special execution (`dragon pytest -s ...` instead of `pytest -s ...` We don't want to execute all tests in the dragon environment, so dragon tests remain separate.

File: pyproject.toml (hunk 17 above)
Comment (MattToast) [reply pyproject.toml:148]: > we should get this as-is into v1.0 and work on making mypy pass to get v1.0 into develop

Agreed! These suppressions are largely code that has become unused in the transition to V1. In theory, most these modules will end up being deleted or significantly reworked, and we just didn't want to lose velocity trying to make static analysis pass on something that we know we will not use.

TL;DR These should get addressed sometime before we officially move away from the V0.* API being used on `develop`

Comment (ankona) [reply pyproject.toml:148]: added https://github.com/CrayLabs/Poseidon/issues/846

File: doc/changelog.md (hunk 16 above)
Comment (ankona) [reply doc/changelog.md:16]: added https://github.com/CrayLabs/Poseidon/issues/847

**Review by MattToast — APPROVED** [review]
LGTM!! Thanks for getting us down to _one_ really big feature branch instead of _two_ really big (and constantly diverging) feature branches lol

**Review by amandarichardsonn — APPROVED** [review]
Hey bro, gnarly job! Keep shredding and stay stoked, man! LGTM 🌊🏄‍♂️
